Dockerfile.dapper

```diff
@@ -62,19 +59,59 @@ RUN mkdir /usr/tmp && \
     mv /usr/tmp/helm /usr/bin/helm_v3 && \
     chmod +x /usr/bin/kustomize
 
-RUN curl -sLf ${!K3S_DOWNLOAD_URL} -o /usr/bin/k3s && \
-    chmod +x /usr/bin/k3s && \
-    mkdir -p /go/src/github.com/rancher/rancher/.kube && \
+# Set up K3s: copy the necessary binaries from an archive with extracted version of K3s.
+COPY --from=rancher/k3s:v1.22.3-k3s1 \
+    /bin/blkid \
+    /bin/charon \
+    /bin/check-config \
+    /bin/cni \
+    /bin/conntrack \
+    /bin/containerd \
+    /bin/containerd-shim-runc-v2 \
+    /bin/ethtool \
+    /bin/fuse-overlayfs \
+    /bin/ip \
+    /bin/ipset \
+    /bin/losetup \
+    /bin/pigz \
+    /bin/runc \
+    /bin/slirp4netns \
+    /bin/socat \
+    /bin/swanctl \
+    /bin/which \
+    /bin/aux/xtables-legacy-multi \
+/usr/bin/
+
+RUN ln -s /usr/bin/cni /usr/bin/bridge && \
+    ln -s /usr/bin/cni /usr/bin/flannel && \
+    ln -s /usr/bin/cni /usr/bin/host-local && \
+    ln -s /usr/bin/cni /usr/bin/loopback && \
+    ln -s /usr/bin/cni /usr/bin/portmap && \
+    ln -s /usr/bin/containerd /usr/bin/crictl && \
+    ln -s /usr/bin/containerd /usr/bin/ctr && \
+    ln -s /usr/bin/containerd /usr/bin/k3s && \
+    ln -s /usr/bin/containerd /usr/bin/k3s-agent && \
+    ln -s /usr/bin/containerd /usr/bin/k3s-etcd-snapshot && \
+    ln -s /usr/bin/containerd /usr/bin/k3s-server && \
+    ln -s /usr/bin/containerd /usr/bin/kubectl && \
+    ln -s /usr/bin/pigz /usr/bin/unpigz && \
+    ln -s /usr/bin/xtables-legacy-multi /usr/bin/iptables && \
+    ln -s /usr/bin/xtables-legacy-multi /usr/bin/iptables-save && \
+    ln -s /usr/bin/xtables-legacy-multi /usr/bin/iptables-restore && \
+    ln -s /usr/bin/xtables-legacy-multi /usr/bin/iptables-translate && \
+    ln -s /usr/bin/xtables-legacy-multi /usr/bin/ip6tables && \
+    ln -s /usr/bin/xtables-legacy-multi /usr/bin/ip6tables-save && \
+    ln -s /usr/bin/xtables-legacy-multi /usr/bin/ip6tables-restore && \
+    ln -s /usr/bin/xtables-legacy-multi /usr/bin/ip6tables-translate
+
+RUN mkdir -p /go/src/github.com/rancher/rancher/.kube && \
     ln -s /etc/rancher/k3s/k3s.yaml /go/src/github.com/rancher/rancher/.kube/k3s.yaml
 
 RUN curl -sLf https://github.com/rancher/k3s/releases/download/${CATTLE_K3S_VERSION}/k3s-images.txt -o /usr/tmp/k3s-images.txt
 
 ENV YQ_URL=https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_${ARCH}
 RUN curl -sLf ${YQ_URL} -o /usr/bin/yq && chmod +x /usr/bin/yq
 
-ENV KUBECTL_URL=https://storage.googleapis.com/kubernetes-release/release/v1.22.3/bin/linux/${ARCH}/kubectl
```

Comment:
If I understand this correctly, the kubectl binary can be removed here because it's packaged in the k3s image. Did you test that it still works?

Reply:
That's right. It works. I will test again with a few other binaries removed and also test using kubectl directly inside the container.

Reply:
Ok great! Can you also check you can ping embedded etcd?

Reply:
You can do that using this doc https://github.com/rancher/rancher/wiki/How-to-Use-Etcdctl-with-K3s-Embedded-Etcd. So basically run this cmd

etcdctl --endpoints=https://localhost:2379 \
          --cert="/var/lib/rancher/k3s/server/tls/etcd/server-client.crt" \
          --key="/var/lib/rancher/k3s/server/tls/etcd/server-client.key" \
          --cacert="/var/lib/rancher/k3s/server/tls/etcd/server-ca.crt" member list

And make sure you get a cluster ID and details.